sys/net/application_layer/gcoap/gcoap.c

```diff
@@ -732,6 +881,95 @@ static void _find_obs_memo_resource(gcoap_observe_memo_t **memo,
     }
 }
 
+/*
+ * Transport layer functions
+ */
+
+static void _tl_get_socket(coap_socket_t *sock)
+{
+#if IS_ACTIVE(CONFIG_GCOAP_ENABLE_DTLS)
+    sock->type = COAP_SOCKET_TYPE_DTLS;
+    sock->socket.dtls = &_sock_dtls;
+#else
+    sock->type = COAP_SOCKET_TYPE_UDP;
+    sock->socket.udp = &_sock_udp;
+#endif
+}
+
+static ssize_t _tl_send(coap_socket_t *sock, const void *data, size_t len,
+                      const sock_udp_ep_t *remote)
+{
+    int res;
+#if IS_ACTIVE(CONFIG_GCOAP_ENABLE_DTLS)
```

Comment:
We should probably also check `sock->type` here so a DTLS enabled client can still send unencrypted requests

Reply:
True. Theoretically right now, we could send unencrypted...but AFAIK we "cannot" receive unencrypted. (We can, but this would just lead to encryption/DTLS errors, since DTLS headers are expected)

Reply:
Hm what would be needed to also being able to receive non-encrypted answers?
We could do that as a follow-up, but generally it should still be possible to do unencrypted requests when connecting to a non-COAPS server

Reply:
1) gcoap API change to distinguish between chosen transport types
2) Another UDP socket for running the non-COAPS server.
That should be enough to make it work.

Reply:
That would mean that we expect to communicate with RIOT nodes in the default configuration. 5684 may be the default for gcoap over DTLS, but it is not mandatory.